microsoft-edge/webview2/how-to/winrt-from-js.md

```diff
@@ -0,0 +1,432 @@
+---
+title: Call native-side WinRT code from web-side code
+description: How to pass WinRT host objects to JavaScript using the AddHostObjectToScript API for WebView2 apps.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: webview
+ms.date: 05/03/2022
+---
+# Call native-side WinRT code from web-side code
+
+Your web-side JavaScript code can access native-side WinRT methods and properties, with the help of the **wv2winrt** tool (the WebView2 WinRT JS Projection tool).  The **wv2winrt** tool generates needed code files for your JavaScript code, and enables using methods and properties of any WinRT APIs, including:
+
+*  Your WebView2 host app's WinRT APIs.
+*  Windows WinRT APIs.
+*  Third-party WinRT APIs.
+
+For more information about why you'd want to have your web-side JavaScript code access the methods and properties of your WinRT host app, see the introduction of [Call native-side code from web-side code](hostobject.md).
+
+This article walks you through the following main steps:
+
+1. Create a **WinRTAdapter** project for the wv2winrt tool.
+
+1. Run the wv2winrt tool to generate C++/WinRT source code for selected namespaces or classes.
+
+1. Call [AddHostObjectToScript](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2#addhostobjecttoscript), in the WinUI 2 (UWP) sample WebView2 browser app project, [WebView2Samples > webview2_sample_uwp](https://github.com/MicrosoftEdge/WebView2Samples/tree/main/SampleApps/webview2_sample_uwp).
+
+1. Call methods and properties on the host object from your web-side JavaScript code (or from the DevTools Console).
+
+
+##### Why WinRT and .NET use different approaches
+
+This article is for WinRT WebView2 APIs, not for .NET WebView2 APIs.  The C# code in this article will build, but not run, for .NET WebView2 APIs.  Calling `AddHostObjectToScript` using this article's C# code for .NET WebView2 APIs would produce an error message.
+
+The wv2winrt tool (the WebView2 WinRT JS Projection tool) is needed when projecting WinRT objects, because WinRT doesn't support `IDispatch` or any other mechanism to dynamically inspect and interact with WinRT objects, which WebView2's Win32 and .NET platforms support.  For .NET use of `AddHostObjectToScript`, see [Call native-side code from web-side code](hostobject.md) instead of this article.
+<!--
+`IDispatch` is a way to:
+*  Dynamically inspect COM objects, to discover methods, properties, and events.
+*  Interact with COM objects, to call methods, get or set properties, and subscribe to and receive events.
+-->
+
+
+<!-- ====================================================================== -->
+## Let's get started!
```

Comment:
Consider changing to 'Objective, or End goal' or something like that. We are not getting started yet here, because you're still giving an overview in the paragraph below.

Reply:
```
## End goal of this example

First, we'll pick some WinRT APIs that we're interested in calling from JavaScript code.
```